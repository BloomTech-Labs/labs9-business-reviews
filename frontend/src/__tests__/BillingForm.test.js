import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';
import BillingFormComponent from '../components/BillingForm';

describe('<BillingFormComponent />', () => {
  it('renders', () => {
    shallow(<BillingFormComponent />);
  });
  it('matches snapshot', () => {
    const wrapper = shallow(<BillingFormComponent />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it('should contain 1 div', () => {
    let app = shallow(<BillingFormComponent />);
    expect(app.find("div").length).toBe(1);
  });
  it('should contain 1 p tag', () => {
    let app = shallow(<BillingFormComponent />);
    expect(app.find("p").length).toBe(1);
  });
  it('should contain 1 h1 element', () => {
    let app = shallow(<BillingFormComponent />);
    expect(app.find("h1").length).toBe(1);
    expect(app.find('h1').text()).toMatch("Bonafind Subscription");
  });
  it('should contain 1 form', () => {
    let app = shallow(<BillingFormComponent />);
    expect(app.find("form").length).toBe(1);
  });
  it('should contain 2 inputs', () => {
    let app = shallow(<BillingFormComponent />);
    expect(app.find("input").length).toBe(2);
  });
});
