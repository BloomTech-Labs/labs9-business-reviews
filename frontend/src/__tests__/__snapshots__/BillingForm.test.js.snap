// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BillingFormComponent /> matches snapshot 1`] = `
<GatedSignIn>
  <styled.div>
    <div
      className="billing-form"
    >
      <h1>
        Bonafind Subscription
      </h1>
      <p>
        Here you can choose between a monthly or yearly subscription. Without a subscription, you can create 3 reviews. But with a subscription you can write as many reviews as you like!
      </p>
      <form
        className="radio-form"
      >
        <input
          name="subscription"
          onClick={[Function]}
          type="radio"
          value="yearly"
        />
        1 Year Subscription - $9.99 
        <br />
         
        <br />
        <input
          name="subscription"
          onClick={[Function]}
          type="radio"
          value="monthly"
        />
        1 Month Subscription - $0.99 
        <br />
      </form>
      <ReactStripeCheckout
        ComponentClass="span"
        amount={null}
        className="StripeCheckout"
        currency="USD"
        description="Purchase Subscription"
        label="Pay With Card"
        locale="auto"
        name="Bonafind"
        reconfigureOnUpdate={false}
        stripeKey="pk_test_YRDXagNKMjZOXlX2ULVNUWbT"
        token={[Function]}
        triggerEvent="onClick"
      />
    </div>
  </styled.div>
</GatedSignIn>
`;
